### PR TITLE
refactor: update `blas/ext/base/scusumpw` to follow current project conventions

### LINKED ISSUE #1520: [RFC]: refactor `blas/ext/base/scusumpw` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/scusumpw`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/scusumpw) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by rxbryan**
updates blas/ext/base/scusumpw to follow current project conventions

## Related Issues
 
> Does this pull request have any related issues?
Yes
Issue #788 
Issue #1152 
 
This pull request:

-   resolves part of #1152 
-   fixes #1520

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/ext/base/scusumpw/lib/ndarray.native.js [hunk 1]
```diff
@@ -53,14 +54,12 @@ var addon = require( './scusumpw.native.js' );
...
-	if ( strideY < 0 ) {
-		offsetY += (N-1) * strideY;
-	}
-	viewX = new Float32Array( x.buffer, x.byteOffset+(x.BYTES_PER_ELEMENT*offsetX), x.length-offsetX ); // eslint-disable-line max-len
-	viewY = new Float32Array( y.buffer, y.byteOffset+(y.BYTES_PER_ELEMENT*offsetY), y.length-offsetY ); // eslint-disable-line max-len
+	offsetX = minViewBufferIndex( N, strideX, offsetX );
+	offsetY = minViewBufferIndex( N, strideY, offsetY );
+
+	viewX = offsetView( x, offsetX );
+	viewY = offsetView( y, offsetY );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/scusumpw/lib/ndarray.native.js:62]:
```suggestion

	offsetX = minViewBufferIndex( N, strideX, offsetX );
	offsetY = minViewBufferIndex( N, strideY, offsetY );

	viewX = offsetView( x, offsetX );
	viewY = offsetView( y, offsetY );
```

**Review by Pranavchiku — APPROVED** [review]
Looks good to me, thanks @rxbryan !

**Comment by rxbryan** [comment]
Hi @Pranavchiku. whats the status of the pr?

File: lib/node_modules/@stdlib/blas/ext/base/scusumpw/src/addon.c [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+*
+* -   When called from JavaScript, the function expects six arguments:
+*
+*     -   `N`: number of indexed elements
+*     -   `sum`: initial sum
+*     -   `X`: input array
+*     -   `strideX`: `X` stride length
+*     -   `Y`: output array
+*     -   `strideY`: `Y` stride length
+*/
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumpw/src/addon.c]:
```suggestion
/**
* Receives JavaScript callback invocation data.
*
* @private
* @param env    environment under which the function is invoked
* @param info   callback data
* @return       Node-API value
*/
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumpw/src/addon.c [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+static napi_value addon( napi_env env, napi_callback_info info ) {
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 6 );
+	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+	STDLIB_NAPI_ARGV_FLOAT( env, sum, argv, 1 );
+	STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 3 );
+	STDLIB_NAPI_ARGV_INT64( env, strideY, argv, 5 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, strideX, argv, 2 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, Y, N, strideY, argv, 4 );
+
+	stdlib_strided_scusumpw( N, (float)sum, (float *)X, strideX, (float *)Y, strideY );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumpw/src/addon.c]: Casting is already taken care of now with the macros:

```suggestion
	stdlib_strided_scusumpw( N, sum, X, strideX, Y, strideY );
```

**Review by Planeshifter — APPROVED** [review]
Thanks for the PR! Will merge once CI passes after a few final changes.

**Pull request merged**